infrastructure/code/code.go

```diff
@@ -189,9 +191,10 @@ func (sc *Scanner) Scan(ctx context.Context, path string, folderPath string) (is
 	results, err := internalScan(ctx, sc, folderPath, logger, filesToBeScanned)
 
 	if err == nil && c.IsDeltaFindingsEnabled() {
-		err = scanAndPersistBaseBranch(ctx, sc, folderPath)
-		if err != nil {
-			logger.Error().Err(err).Msg("couldn't scan base branch for folder " + folderPath)
+		baseScanErr := scanAndPersistBaseBranch(ctx, sc, folderPath)
+		if baseScanErr != nil {
+			go sc.notifier.SendShowMessage(sglsp.MTError, "Couldn't determine the diff between current and base branch. Falling back to showing full scan results.")
```

Comment:
You don't really need a goroutine here, but doesn't hurt either. Usually, it's only necessary to use a goroutine, when we're waiting for an answer, e.g. with `ShowMessageRequest`